tests.py

```diff
@@ -0,0 +1,77 @@
+        
+        s=0.4        
+        
+        results = [param_calc.lerpN(s,p) for p in self.ps]
+        
+        self.assertAlmostEqual(results, expected)
+        
+    def test_lerpN_075(self):
+        expected = [0.0,2.562987316014611,3.3813610284845486,3.957876423389526,4.119592358163913,4.410381026883067,4.542144081315797,4.783253988536195,4.894039286852489,5.098841470385992,5.193728353401936,5.284087964381649,5.3702241145612595,5.452403629130475,5.530862229358699,5.605809260032427,5.6774315271544555,5.745896441918097,5.811354617919749,5.873942033151851,5.933781842388968,5.990985906360208,6.045656089689916,6.097885368666876,6.147758781542594,6.195354247601323,6.240743276212692,6.283991583125086,6.325159628129639,6.364303085730988,6.401473258459592,6.4700792407495005,6.501598863711478,6.559256873566736,6.585460790887023,6.632762864557223,6.653911686416458,6.707608520188385,6.747043602033591,6.7619973862189955]
+        
+        s=0.75        
+        
+        results = [param_calc.lerpN(s,p) for p in self.ps]
+        
+        self.assertAlmostEqual(results, expected)
+        
+    def test_lerpN_01377(self):
+        expected = [0.0,656.2091188634356,4430.465017638806,13111.539212851487,17278.968237700436,27648.332190334,33867.29795830724,48371.576356735204,56640.022663156175,75132.40961385205,85313.11207904678,96081.93232098296,107408.64949582303,119260.1712654393,131600.82767280727,144392.65221480723,157595.65097524057,171168.06055235,185066.59541068293,199246.6852014968,213662.70252492183,228268.18154786233,243016.02784172457,257858.71976177508,272748.50165381265,287637.569142803,302478.2467313146,317223.15791225695,331825.3879800286,346238.6397062968,360417.3820307715,387893.8894012034,401105.66623427684,426270.80846425035,438146.2814065397,460300.3000122972,470510.96006053337,497309.95294645196,517812.61710336444,525773.9573645486]
+        
+        s=0.1377        
+        
+        results = [param_calc.lerpN(s,p) for p in self.ps]
+        
+        self.assertAlmostEqual(results, expected)
+        
+    def test_slope(self):
+        expected = [0.75,0.13768115942028955,0.5,0.5,0.49999999999999994,0.49999999999999994,0.0,0.5000000000000001,0.5,0.4999999999999999,0.5,0.5,0.5000000000000001,0.5,0.4999999999999996,0.5000000000000001,0.5000000000000004,0.5000000000000006,0.6000000000000001,0.4999999999999999,0.5000000000000001,0.0,0.5,0.5,0.5,0.4999999999999999,0.5,0.5,0.4999999999999996,0.5,0.5000000000000001,0.5,0.5,0.5,0.5,0.5,0.6000000000000001,0.5,0.4999999999999999]
```

Comment:
shouldnt we round the 0.4999... to 0.5? because rn this looks like auto generated instead of done by hand (which we should do for unit tests)

Reply:
They are indeed automatically generated, from the output of each function on e3a563b80646232b788b856698f8f429417052e1. After all, the tests weren't written before the code - like they would have in TDD; the intention has been to ensure the output remains unchanged after future refactoring.

Generally speaking, I think parameterised unit testing would be called for.

For what is worth,  the generated numbers for ```lerpN``` were checked against the dictionaries of memory-read values (for s=0.6 and s=0.4 ).